sound/soc/codecs/max98373.c

```diff
@@ -168,6 +168,23 @@ static SOC_ENUM_SINGLE_DECL(max98373_adc_samplerate_enum,
 			    MAX98373_R2051_MEAS_ADC_SAMPLING_RATE, 0,
 			    max98373_ADC_samplerate_text);
 
+static int max98373_feedback_get(struct snd_kcontrol *kcontrol,
+				 struct snd_ctl_elem_value *ucontrol)
+{
+	struct snd_soc_component *component = snd_kcontrol_chip(kcontrol);
+
+	if (snd_soc_component_get_bias_level(component) == SND_SOC_BIAS_OFF) {
+		/*
+		 * We can't get the feedback data when codec is off, set
+		 * value = 0 and return 0 to avoid error.
+		 */
+		ucontrol->value.integer.value[0] = 0;
```

Comment:
can we set value to 0xFFFFFFF make user be aware that it is invalid ?

Reply:
> can we set value to 0xFFFFFFF make user be aware that it is invalid ?

I am fine with that. What is your opinion @ryans-lee ?

Reply:
Do you mean 0xFFFFFFF not 0xFFFFFFFF? Either ways are okay to me.

Reply:
@ryans-lee What does the value mean? My purpose is that user application will do nothing if it gets the value when codec is suspended. Which value is better to you? 0 or 0xFFFFFFFF? BTW ucontrol->value.integer.value[0] is long type, so we can set it to -1 if we want.

Reply:
@bardliao I've been reading through the previous PRs and it is starting to seem this may be the best option. One thing that still bugs me is that what user-space app is actually reading the values when the problem occurs. I did not find this info in https://github.com/thesofproject/sof/issues/3541 for instance. How would the user-space application filter valid values from non-valid ones? Benefit of -EBUSY is that it's at least a well-documented interface and user-space knows it can't get the value now.

Reply:
I'd agree with @kv2019i - let's not re-invent the wheel. If `.cache_only` is enabled, `regmap_read()` returns `-EBUSY`. This isn't the only codec driver that uses regmap and supports rt-pm, is it? What do other drivers do? How do other user-space programs handle these cases?

Reply:
> I'd agree with @kv2019i - let's not re-invent the wheel. If `.cache_only` is enabled, `regmap_read()` returns `-EBUSY`. This isn't the only codec driver that uses regmap and supports rt-pm, is it? What do other drivers do? How do other user-space programs handle these cases?

`regmap_read()` returns `-EBUSY` because `.cache_only` is enabled AND the register is volatile. Using a volatile register in a kcontrol is not common. I think using specific _get/_put functions is a common way to handle a kcontrol which is not just simply read/write a register bit to get/set the kcontrol value.